openmc/mgxs/mgxs.py

```diff
@@ -4615,3 +4628,728 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
+                # we can use the generic get_xs(...) method
+                xs_tally = prompt_nu_fission_out / prompt_nu_fission_in
+
+                # Add the coarse energy filter back to the nu-fission tally
+                prompt_nu_fission_in.filters.append(energy_filter)
+
+                xs = xs_tally.get_values(filters=filters,
+                                         filter_bins=filter_bins, value=value)
+
+            # Get chi prompt for all nuclides in the domain
+            elif nuclides == 'all':
+                nuclides = self.get_all_nuclides()
+                xs = self.xs_tally.get_values(filters=filters,
+                                              filter_bins=filter_bins,
+                                              nuclides=nuclides, value=value)
+
+            # Get chi prompt for user-specified nuclides in the domain
+            else:
+                cv.check_iterable_type('nuclides', nuclides, basestring)
+                xs = self.xs_tally.get_values(filters=filters,
+                                              filter_bins=filter_bins,
+                                              nuclides=nuclides, value=value)
+
+        # If chi prompt was computed as an average of nuclides in the domain
+        else:
+            xs = self.xs_tally.get_values(filters=filters,
+                                          filter_bins=filter_bins, value=value)
+
+        # Reverse data if user requested increasing energy groups since
+        # tally data is stored in order of increasing energies
+        if order_groups == 'increasing':
+
+            # Reshape tally data array with separate axes for domain and energy
+            if groups == 'all':
+                num_groups = self.num_groups
+            else:
+                num_groups = len(groups)
+            num_subdomains = int(xs.shape[0] / num_groups)
+            new_shape = (num_subdomains, num_groups) + xs.shape[1:]
+            xs = np.reshape(xs, new_shape)
+
+            # Reverse energies to align with increasing energy groups
+            xs = xs[:, ::-1, :]
+
+            # Eliminate trivial dimensions
+            xs = np.squeeze(xs)
+            xs = np.atleast_1d(xs)
+
+        xs = np.nan_to_num(xs)
+        return xs
+
+
+class Velocity(MGXS):
+    """A velocity multi-group cross section.
+
+    This class can be used for both OpenMC input generation and tally data
+    post-processing to compute spatially-homogenized and energy-integrated
+    multi-group velocity cross sections for multi-group neutronics
+    calculations. At a minimum, one needs to set the
+    :attr:`Velocity.energy_groups` and :attr:`Velocity.domain`
+    properties. Tallies for the flux and appropriate reaction rates over the
+    specified domain are generated automatically via the
+    :attr:`Velocity.tallies` property, which can then be appended to a
+    :class:`openmc.Tallies` instance.
+
+    For post-processing, the :meth:`MGXS.load_from_statepoint` will pull in the
```

Comment:
Sure, I'll specialize the docstring. The velocity is actually computed by first computing the flux weighted inverse velocity and then taking the inverse of that so more detail would probably be good to clarify that.

The units of velocity are cm/s. I'll note that here. While m/s seems like a more common unit for velocity, I chose to keep it as cm/s since the flux and cross sections all use cm.